rocksdb.nim

```diff
@@ -114,17 +103,49 @@ template getImpl(T: type) {.dirty.} =
   else:
     result = err("")
 
-proc get*(db: RocksDBInstance, key: KeyValueType): RocksDBResult[string] =
+proc get*(db: RocksDBInstance, key: openArray[byte], onData: DataProc): RocksDBResult[bool] =
+  if key.len <= 0:
+    return err("rocksdb: key cannot be empty on get")
+
+  var
+    errors: cstring
+    len: csize_t
+    data = rocksdb_get(db.db, db.readOptions,
+                       cast[cstring](unsafeAddr key[0]), csize_t(key.len),
+                       addr len, addr errors)
+  bailOnErrors()
+  if not data.isNil:
```

Comment:
How about wrapping the whole `if` with a `try/finally`. This won't leak memory even if `onData` exits with a Defect.

Reply:
meh - that would cause it to have to pay the exception complexity price (setjmp and rtti) - at that point, might as well make it fully exception safe and allow exceptions through the callback

Reply:
i think we should operate under the assumption that we'll be compiling the retail code with C++.

Reply:
which also pays the rtti cost on exception - double afair, c++ and nim. I'm not completely opposed but I'd leave it for a later commit when these things have been clean up a bit more and we can assess the impact better.

Reply:
A TODO comment may be appropriate then.

Reply:
pushed